components/brave_news/browser/peeking_card_unittest.cc

```diff
@@ -0,0 +1,145 @@
+
+ArticleInfo FakeInfo(std::string id = "",
+                     std::vector<std::string> channels = {},
+                     GURL url = GURL()) {
+  static int next_id = 0;
+
+  if (id.empty()) {
+    id = base::NumberToString(++next_id);
+  }
+
+  auto item = mojom::FeedItemMetadata::New();
+  item->publisher_id = id;
+  item->channels = channels;
+  item->url = url;
+
+  return std::make_tuple(std::move(item), ArticleMetadata());
+}
+
+}  // namespace
+
+TEST(BraveNewsPeekingCard, NoArticles) {
+  EXPECT_EQ(std::nullopt, PickPeekingCard(SubscriptionsSnapshot(), {}, {}));
+}
+
+TEST(BraveNewsPeekingCard, NoSubscribedArticles) {
+  ArticleInfos articles;
+  articles.push_back(FakeInfo());
+  articles.push_back(FakeInfo());
+  articles.push_back(FakeInfo());
+  EXPECT_EQ(std::nullopt,
+            PickPeekingCard(SubscriptionsSnapshot(), {}, articles));
+}
+
+TEST(BraveNewsPeekingCard, PublishersAreIncluded) {
+  ArticleInfos articles;
+  articles.push_back(FakeInfo());
+  articles.push_back(FakeInfo());
+  articles.push_back(FakeInfo());
+
+  size_t enabled_index = 1;
+  SubscriptionsSnapshot subscriptions(
+      {std::get<0>(articles[enabled_index])->publisher_id}, {}, {}, {});
+
+  EXPECT_EQ(enabled_index,
+            PickPeekingCardWithMax(subscriptions, {}, articles, 1));
+}
+
+TEST(BraveNewsPeekingCard, ChannelsAreIncluded) {
+  ArticleInfos articles;
+  articles.push_back(FakeInfo());
+  articles.push_back(FakeInfo("foo", {"one"}));
+  articles.push_back(FakeInfo());
+
+  base::flat_map<std::string, std::vector<std::string>> channels;
+  channels["en_NZ"] = {"one"};
+  SubscriptionsSnapshot subscriptions({}, {}, {}, channels);
+
+  EXPECT_EQ(1, PickPeekingCardWithMax(subscriptions, {}, articles, 1));
+}
+
+TEST(BraveNewsPeekingCard, DirectFeedsAreIncluded) {
+  ArticleInfos articles;
+  articles.push_back(FakeInfo());
+  articles.push_back(FakeInfo("foo"));
+  articles.push_back(FakeInfo());
+
+  std::vector<DirectFeed> feeds;
+  feeds.push_back({"foo", GURL(), ""});
+  SubscriptionsSnapshot subscriptions({}, {}, feeds, {});
+
+  EXPECT_EQ(1, PickPeekingCardWithMax(subscriptions, {}, articles, 1));
+}
+
+TEST(BraveNewsPeekingCard, DisabledPublishersExcluded) {
+  ArticleInfos articles;
+  articles.push_back(FakeInfo("bar"));
+  articles.push_back(FakeInfo("foo"));
+  articles.push_back(FakeInfo("frob"));
+
+  SubscriptionsSnapshot subscriptions({"foo"}, {"foo", "bar", "frob"}, {}, {});
+
+  EXPECT_EQ(std::nullopt,
+            PickPeekingCardWithMax(subscriptions, {}, articles, 1));
+}
+
+TEST(BraveNewsPeekingCard, DirectFeedsAreHigherThanPublishers) {
+  ArticleInfos articles;
+  articles.push_back(FakeInfo("combined"));
+  articles.push_back(FakeInfo("direct"));
+  articles.push_back(FakeInfo("other"));
+
+  std::vector<DirectFeed> feeds;
+  feeds.push_back({"direct", GURL(), ""});
+  SubscriptionsSnapshot subscriptions({"other", "combined"}, {}, feeds, {});
+
+  EXPECT_EQ(1, PickPeekingCardWithMax(subscriptions, {}, articles, 1));
```

Comment:
`1` here means it's from direct feed?
Can't know picked card is direct or not in this test case.
All below test cases are same.

Reply:
Ah I see. `1` in `EXPECT_EQ(1, ...` is the direct article index in `articles`.